### PR TITLE
Print events on cleanup

### PR DESCRIPTION
**Pull request by unguiculus**
Helps identifying issues when Helm timeout because pods cannot be created due to errors.

Affected PR: https://github.com/helm/charts/pull/13444

### CODE REVIEW

File: pkg/chart/chart.go [hunk 1]
```diff
@@ -812,7 +815,13 @@ func (t *Testing) ValidateMaintainers(chart *Chart) error {
 	return nil
 }
 
-func (t *Testing) PrintPodDetailsAndLogs(namespace string, selector string) {
+func (t *Testing) PrintEventsPodDetailsAndLogs(namespace string, selector string) {
+	util.PrintDelimiterLine("=")
+
+	printDetails(namespace, "Events of namespace", ".", func(item string) error {
+		return t.kubectl.GetEvents(namespace)
```
Comment (jlegrone) [inline pkg/chart/chart.go]: Is it possible to use a label selector for events?

Comment (unguiculus) [reply pkg/chart/chart.go]: Yes, that's possible. I guess that might be better if someone uses a pre-defined namespace

Comment (unguiculus) [reply pkg/chart/chart.go]: Actually, that won't quite work because the events won't have the labels of the chart resources.

Comment (unguiculus) [reply pkg/chart/chart.go]: You could do something like this:

`kubectl get events --field-selector 'involvedObject.name=foo'`

But that could be tricky. Depending on resource naming we might miss something.

**Comment by scottrigby** [comment]
I like this 👍

File: pkg/chart/chart.go (hunk 1 above)
Comment (jlegrone) [reply pkg/chart/chart.go]: 😞that's too bad... as long as tons of existing events doesn't bog down performance or place unnecessary load on the api server this seems okay for either case though.

File: pkg/tool/kubectl.go [hunk 2]
```diff
@@ -184,6 +184,10 @@ func (k Kubectl) GetPods(args ...string) ([]string, error) {
 	return strings.Fields(pods), nil
 }
 
+func (k Kubectl) GetEvents(namespace string) error {
+	return k.exec.RunProcess("kubectl", "get", "events", "--namespace", namespace)
```
Comment (jlegrone) [inline pkg/tool/kubectl.go]: Adding `-o wide` also includes a `COUNT` column which could be useful.

```suggestion
	return k.exec.RunProcess("kubectl", "get", "events", "--output", "wide", "--namespace", namespace)
```

File: pkg/chart/chart.go (hunk 1 above)
Comment (unguiculus) [reply pkg/chart/chart.go]: I dont see an issue here.

**Comment by kpucynski** [comment]
It seems that this PR broke master build:
```
$ ./build.sh 
# chart-testing/pkg/chart [chart-testing/pkg/chart.test]
pkg/chart/chart.go:252:3: cannot use tool.NewKubectl(procExec) (type tool.Kubectl) as type Kubectl in field value:
	tool.Kubectl does not implement Kubectl (missing GetEvents method)
?   	chart-testing/ct	[no test files]
?   	chart-testing/ct/cmd	[no test files]
FAIL	chart-testing/pkg/chart [build failed]
ok  	chart-testing/pkg/config	0.002s
?   	chart-testing/pkg/exec	[no test files]
ok  	chart-testing/pkg/tool	0.011s
ok  	chart-testing/pkg/util	0.006s
FAIL
$ go version go1.13.1 linux/amd64
```

**Pull request merged**